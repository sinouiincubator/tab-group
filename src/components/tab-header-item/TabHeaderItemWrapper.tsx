import styled, { css } from 'styled-components';

const hoverCss = css`
  &:hover {
    opacity: 1;
  }
`;

const activeCss = css`
  color: ${(props) => props.theme.palette.primary[500]};
  opacity: 1;
`;

const disabledCss = css`
  cursor: not-allowed;
  color: ${(props) => props.theme.palette.text.disabled};
`;

interface TabHeaderItemWrapperProps {
  disabled?: boolean;
  active?: boolean;
}

const TabHeaderItemWrapper = styled.div<TabHeaderItemWrapperProps>`
  height: 48px;
  padding: 0 24px;
  cursor: pointer;
  box-sizing: border-box;
  opacity: 0.6;
  text-align: center;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  white-space: nowrap; /* 不换行显示 */
  color: ${(props) => props.theme.palette.text.primary};
  outline: none;
  user-select: none;

  letter-spacing: 1.25px;
  font-weight: 500;

  position: relative;

  transition: opacity 0.25s cubic-bezier(0.4, 0, 0.2, 1),
    color 0.25s cubic-bezier(0.4, 0, 0.2, 1);

  & > .sinoui-tab-label-content {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    white-space: nowrap;
  }

  ${(props) => !props.disabled && hoverCss};
  ${(props) => props.disabled && disabledCss};
  ${(props) => props.active && activeCss};
`;

export default TabHeaderItemWrapper;
