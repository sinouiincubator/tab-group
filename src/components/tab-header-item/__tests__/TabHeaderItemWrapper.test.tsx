import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import 'jest-styled-components';
import TabHeaderItemWrapper from '../TabHeaderItemWrapper';

it('正确渲染TabHeaderItemWrapper', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderItemWrapper />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('选中状态下的TabTabHeaderItemWrapper', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderItemWrapper active />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('不可用状态下的TabHeaderItemWrapper', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderItemWrapper disabled />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('不可用且选中状态下的TabHeaderItemWrapper', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <TabHeaderItemWrapper disabled active />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
