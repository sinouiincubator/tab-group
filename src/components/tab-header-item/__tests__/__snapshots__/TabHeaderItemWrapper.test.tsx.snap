// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`不可用且选中状态下的TabHeaderItemWrapper 1`] = `
.c0 {
  height: 48px;
  padding: 0 24px;
  cursor: pointer;
  box-sizing: border-box;
  opacity: 0.6;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  color: #3f51b5;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  font-weight: 500;
  position: relative;
  white-space: nowrap;
  -webkit-transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  -webkit-tap-highlight-color: transparent;
  cursor: not-allowed;
  opacity: 1;
}

<div
  className="c0"
  disabled={true}
/>
`;

exports[`不可用状态下的TabHeaderItemWrapper 1`] = `
.c0 {
  height: 48px;
  padding: 0 24px;
  cursor: pointer;
  box-sizing: border-box;
  opacity: 0.6;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  color: rgba(0,0,0,0.38);
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  font-weight: 500;
  position: relative;
  white-space: nowrap;
  -webkit-transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  -webkit-tap-highlight-color: transparent;
  cursor: not-allowed;
}

<div
  className="c0"
  disabled={true}
/>
`;

exports[`正确渲染TabHeaderItemWrapper 1`] = `
.c0 {
  height: 48px;
  padding: 0 24px;
  cursor: pointer;
  box-sizing: border-box;
  opacity: 0.6;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  color: rgba(0,0,0,0.87);
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  font-weight: 500;
  position: relative;
  white-space: nowrap;
  -webkit-transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  -webkit-tap-highlight-color: transparent;
}

.c0:hover {
  opacity: 1;
}

<div
  className="c0"
/>
`;

exports[`选中状态下的TabTabHeaderItemWrapper 1`] = `
.c0 {
  height: 48px;
  padding: 0 24px;
  cursor: pointer;
  box-sizing: border-box;
  opacity: 0.6;
  text-align: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  color: #3f51b5;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-letter-spacing: 1.25px;
  -moz-letter-spacing: 1.25px;
  -ms-letter-spacing: 1.25px;
  letter-spacing: 1.25px;
  font-weight: 500;
  position: relative;
  white-space: nowrap;
  -webkit-transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  transition: opacity 0.25s cubic-bezier(0.4,0,0.2,1), color 0.25s cubic-bezier(0.4,0,0.2,1);
  -webkit-tap-highlight-color: transparent;
  opacity: 1;
}

.c0:hover {
  opacity: 1;
}

<div
  className="c0"
/>
`;
